submitit/core/core.py

```diff
@@ -493,6 +493,15 @@ def __del__(self) -> None:
             if not self.watcher.is_done(self.job_id, mode="cache"):
                 self.cancel(check=False)
 
+    def __getstate__(self) -> tp.Dict[str, tp.Any]:
+        return self.__dict__  # for pickling (see __setstate__)
+
+    def __setstate__(self, state: tp.Dict[str, tp.Any]) -> None:
+        """Make sure jobs are registered when loaded from a pickle
+        """
+        self.__dict__.update(state)
+        self.watcher.register_job(self.job_id)
```

Comment:
In __init__ we have:
```
        if not tasks[0]:  # only register for task=0
            self.watcher.register_job(job_id) 
```
Shouldn't it be the same here ? if not please add a comment.
If yes, can we move this into a "_register" method ?

Reply:
done